### DISCUSSION TITLE
many2many ondelete:CASCADE

### DISCUSSION THREAD

**Discussion by ghost** [go-pg/pg#1890] [Q&A] [answered]
If I set `on_delete` to `CASCADE` on a `many2many` relation, for example:

```go
type ModelA struct {
	Id      uint64    `pg:"id"`
	ModelBs []*ModelB `pg:"many2many:modela_to_modelbs,on_delete:CASCADE"`
}

type ModelB struct {
	Id      uint64    `pg:"id"`
	ModelAs []*ModelA `pg:"many2many:modela_to_modelbs,on_delete:CASCADE"`
}

type ModelAToModelB struct {
	ModelAId uint64 `pg:"model_a_id,type:bigint"`
	ModelBId uint64 `pg:"model_a_id,type:bigint"`

        //nolint
	tableName struct{} `pg:"modela_to_modelbs"`
}
```

When I delete a ModelA, will all the ModelB's that are related to that ModelA also be deleted, or just the ModelAToModelB's?

**Comment by vmihailenco** [comment, marked as answer]
`on_delete` only works for has-one relation and it is only used when creating table schema with `CreateTable` . Specifically it adds `FOREIGN_KEY some_id ON DELETE CASCADE` - https://www.postgresql.org/docs/13/sql-createtable.html

This is fixed in Bun by allowing to manually specify foreign keys - https://bun.uptrace.dev/guide/queries.html#creating-tables

**Reply by ghost** [reply]
Okay, thanks for getting back to me. I'll delete the relations manually. :+1:

**Reply by vmihailenco** [reply]
...or you can just update the table definition manually via migrations etc :)
